src/ami-set.js

```diff
@@ -0,0 +1,23 @@
+let base = require('taskcluster-base');
+
+const KEY_CONST = 'ami-set';
+
+ /**
+ * An AMI Set is a collection of AMIs with a single name (its AMI Set ID).
+ * Each AMI in the set is keyed by its virtualization type (PV or HVM) and
+ * by its AWS region.
+ */
+
+let amiSet = base.Entity.configure({
+  version: 1,
+
+  partitionKey: base.Entity.keys.ConstantKey(KEY_CONST),
+  rowKey: base.Entity.keys.StringKey('amiSetId'),
+
+  properties: {
+
+    amiSetId: base.Entity.types.String,
+    amis: base.Entity.types.JSON,
+
+  },
+});
```

Comment:
You get that one for free -- `base.Entity.configure(..)` creates a class for you with a few methods (`create`, `update`, `remove`, etc.).  Sadly there's basically no documentation for that (https://bugzilla.mozilla.org/show_bug.cgi?id=1278385).